apps/nextjs/src/components/AppComponents/download/DownloadAllButton.tsx

```diff
@@ -34,26 +43,35 @@ type DownloadAllButtonProps = {
   downloadAvailable: boolean;
   downloadLoading: boolean;
   data:
-    | {
-        lessonSlides: string | undefined;
-        lessonPlan: string | undefined;
-        worksheet: string | undefined;
-        additionalMaterials: string | undefined;
-        starterQuiz: string | undefined;
-        exitQuiz: string | undefined;
-      }
+    | Data
     | {
         error: unknown;
         message: string;
       }
     | undefined;
-
   "data-testid"?: string;
   chatId: string;
 };
 
 type zipDownloadStatus = "idle" | "loading" | "complete" | "error";
 
+type FileIdsAndFormats = { fileId: string; formats: string[] }[];
+
+const getFileIdsAndFormats = (data: Data): FileIdsAndFormats => {
+  return Object.entries(data)
+    .filter(([, value]) => !!value)
+    .map(([key, value]) => {
+      const fileId = value?.split("/edit")[0]?.split("/d/")?.[1];
+      if (!fileId) {
+        throw new Error("File ID not found");
```

Comment:
Would be nice to add the file id to the error (as a breadcrumb), but perhaps beyond the scope of this fix!

Reply:
Hopefully it's not possible and we just need the assertion to make typescript happy. Hopefully! Otherwise we would want to log out the value as the fileId will obviously be missing

Reply:
Oh ha! Yeah good point, i kinda ignored the context when i made that comment!